spec/ruby/library/stringio/write_spec.rb

```diff
@@ -17,3 +17,23 @@
 describe "StringIO#write when in append mode" do
   it_behaves_like :stringio_write_append, :write
 end
+
+describe "StringIO#write transcoding" do
+  describe "when UTF-16 encoding is set" do
+    it "accepts a UTF-8-encoded string and transcodes it" do
```

Comment:
It would make sense to use a shared spec for this if it's also the behavior for `write_nonblock` or `syswrite`, I will check.

Reply:
It does apply to all 3 `*write*` methods for StringIO, but for IO it's only on `write`. Ah, so inconsistent but at least we already had the correct behavior for `IO`.